crates/polars-plan/src/plans/aexpr/builder.rs

```diff
@@ -40,6 +76,296 @@ impl AExprBuilder {
         }
     }
 
+    pub fn agg(agg: IRAggExpr, arena: &mut Arena<AExpr>) -> Self {
+        Self::new_from_aexpr(AExpr::Agg(agg), arena)
+    }
+
+    pub fn min(self, arena: &mut Arena<AExpr>) -> Self {
+        Self::agg(
+            IRAggExpr::Min {
+                input: self.node(),
+                propagate_nans: false,
+            },
+            arena,
+        )
+    }
+
+    pub fn max(self, arena: &mut Arena<AExpr>) -> Self {
+        Self::agg(
+            IRAggExpr::Min {
```

Comment:
Shouldn't that be IRAggExpr::Max ? I find it odd that max has the same code than min.

Disclaimer: I don't know anything about rust or Polars internal code, so I apologize in advance if this comment is noise.  **IF** this was indeed an oversight, the same comment applies for nan_max below (and your tests are probably not covering this case 😉)